web-app/src/components/HolidayModal/container.js

```diff
@@ -54,49 +56,134 @@ const HolidayModalContainer = Wrapped =>
             title: 'Holiday Approved',
             position: 'top',
           });
+          this.toggleRejectionReasonState();
+          closeModal();
         })
         .catch(error => {
           swal('Error', `There was an error: ${error.message}. 'error`);
         });
     };
 
-    rejectHoliday = () => {
-      const eventId = this.state.holiday.eventId;
+    assignRejectionResponseText = e =>{
+
+      this.setState({ capturedRejectionReponseText: e.target.value });
+    }
+
+    assignRejectionReasonText = e =>{
 
-      rejectHoliday(eventId)
-        .then(() => {
+      this.setState({ capturedRejectionReasonText: e.target.value });
+    }
+
+    toggleRejectionMessageResponse = rejectionReasonToggleValue =>{
+      
+      this.setState({ rejectionReasonResponse: rejectionReasonToggleValue });
+    
+    }
+
+
+    sendRejectionResponse = () => {
+    
+      const rejectionResponseMessage = this.state.capturedRejectionReponseText;
+      const holidayId = this.state.holiday.holidayId;
+   
+      rejectionResponseMessage( holidayId, rejectionResponseMessage )
+        .then( () => {
+        
           this.setState({
             holiday: {
               ...this.state.holiday,
               eventStatus: {
                 ...this.state.holiday.eventStatus,
                 eventStatusId: holidayStatus.REJECTED,
               },
-            },
+            }, 
+          });
+          Toast({
+            type: 'success',
+            title: 'Holiday Rejected',
+            position: 'top',
+          });
+          
+          this.closeModalResetRejectionReasonView();
+          
+        })
+        .catch(error => {
+          swal('Error', `There was an error: ${error.message}. 'error`);
+        });
+
+
+    }
+
+
+    rejectHoliday = () => {
+      
+      const rejectionReasonText = this.state.capturedRejectionReasonText;
+      const holidayId = this.state.holiday.eventId;
+   
+      rejectHolidayWithMessage( holidayId, rejectionReasonText )
+        .then( () => {
+          this.setState({
+            holiday: {
+              ...this.state.holiday,
+              eventStatus: {
+                ...this.state.holiday.eventStatus,
+                eventStatusId: holidayStatus.REJECTED,
+              },
+            }, 
           });
           Toast({
             type: 'success',
             title: 'Holiday Rejected',
             position: 'top',
           });
+          this.closeModalResetRejectionReasonView();
         })
         .catch(error => {
           swal('Error', `There was an error: ${error.message}. 'error`);
         });
     };
 
+
+    toggleRejectionReasonState = () =>{
```

Comment:
a 'toggle' would flip true to false and vice versa - i feel this should be called `collapseRejectionReasonState`

or... if possible combine collapse and reject in a `toggleRejectionReasonState` function doing something along the lines of `this.setState({ !holidayModalExpansion });` - assuming that works, but you get the idea